src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ControllerExtensions.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Data;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Tenants.ViewModels;
+
+namespace OrchardCore.Tenants.Controllers
+{
+    internal static class ControllerExtensions
+    {
+        internal static async Task ValidateModelAsync<TViewModel>(this Controller controller, TViewModel model, bool newTenant = true) where TViewModel : TenantViewModel
+        {
+            var shellHost = controller.HttpContext.RequestServices.GetService<IShellHost>();
+            var featureProfilesService = controller.HttpContext.RequestServices.GetService<IFeatureProfilesService>();
+            var databaseProviders = controller.HttpContext.RequestServices.GetServices<DatabaseProvider>();
+            var S = controller.HttpContext.RequestServices.GetService<IStringLocalizer<TViewModel>>();
+
+            var selectedProvider = databaseProviders.FirstOrDefault(x => x.Value == model.DatabaseProvider);
+
+            if (selectedProvider != null && selectedProvider.HasConnectionString && String.IsNullOrWhiteSpace(model.ConnectionString))
+            {
+                controller.ModelState.AddModelError(nameof(EditTenantViewModel.ConnectionString), S["The connection string is mandatory for this provider."]);
```

Comment:
```suggestion
                controller.ModelState.AddModelError(nameof(TenantViewModel.ConnectionString), S["The connection string is mandatory for this provider."]);
```

and everywhere else, presumably

Reply:
Good catch